turbopack/crates/turbopack-json/src/lib.rs

```diff
@@ -129,9 +129,15 @@ impl EcmascriptChunkItem for JsonChunkItem {
         let data = content.parse_json().await?;
         match &*data {
             FileJsonContent::Content(data) => {
-                let js_str_content = serde_json::to_string(&data.to_string())?;
-                let inner_code = format!("{TURBOPACK_EXPORT_VALUE}(JSON.parse({js_str_content}));");
-
+                let data_str = data.to_string();
+                let inner_code = if data_str.len() > 10_000 {
+                    // Only use JSON.parse if the content is larger than 10kb
```

Comment:
ironically arrays are a case where there is much less of an advantage.  actual cost metrics would be complex, but if there are no objects then the optimization is unlikely to pay for itself.

(annoyingly the blogpost doesn't go into this, but a few years ago i forked the benchmarks and swapped arrays for objects to observe that there was a much smaller delta.)

Still i think this makes sense, just thought you might be curious 😏